src/hoa-rotator.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+
+/**
+ * @fileOverview Sound field rotator for higher-order-ambisonics decoding.
+ */
+
+
+/**
+ * @class Higher-order-ambisonic decoder based on gain node network.
+ *        We expect the order of the channels to conform to ACN ordering.
+ * @param {AudioContext} context    Associated AudioContext.
+ * @param {Number} order            Ambisonic order (2 or 3).
+ */
+function HOARotator (context, order) {
+  this._context = context;
+
+  var num_channels = (order + 1) * (order + 1);
+  this._splitter = this._context.createChannelSplitter(num_channels);
+  this._merger = this._context.createChannelMerger(num_channels);
+
+  // Create a set of per-order rotation matrices using gain nodes.
+  this._m = Array(order);
```

Comment:
Sorry, but we have to spell it out. `m` to `matrices`. Also invoking a constructor without `new` is no t really allowed. Please do this instead: 

```js
this._matrices = [];
```